doc/developer_guide/aws.md

```diff
@@ -0,0 +1,160 @@
+## AWS Build and Release Workflow
+
+The following diagram shows the high-level steps to generate the images:
+
+![image info](./img/create-vm-overview.drawio.png)
+
+### Setup EC2
+
+Creates an EC2 instance based on an Ubuntu AMI via AWS Cloudformation. It also creates a new keypair dynamically.
+After the export has finished, the cloudformation stack and the keypair is removed.
+
+### Install
+
+Installs all dependencies via Ansible:
+* installs Poetry
+* installs and configures Jupyter
+* installs Docker and adds the user `ubuntu` to the docker group
+* clones the script-languages-release repository
+* changes the netplan configuration. This is necessary to have proper network configuration when running the VM image
+
+Finally, the default password will be set, and also the password will be marked as expired, such that the user will be forced to enter a new password during initial login.
+Also, the ssh password authentication will be enabled, and for security reasons the folder "~/.ssh" will be removed.
+
+### Tests
+
+DSS comes with a number of tests in directory `test`.
+There are subdirectories clustering tests with common scope and prerequisites e.g. external resources.
+
+| Directory           | Content |
+|---------------------|---------|
+| `test/unit`         | Simple unit tests requiring no additional setup or external resources. |
+| `test/integration`  | Integration tests with longer runtime and some requiring additional resources. |
+| `test/aws`          | Tests involving AWS resources.  In order to execute these tests you need an AWS account, a user with permissions in this account, and an access key. |
+
+To run the tests in file `test/integration/test_ci.py` please use
+```shell
+export DSS_RUN_CI_TEST=true
+poetry run test/integration/test_ci.py
+```
+
+Creating a docker image is quite time-consuming, currently around 7 minutes. In order to use an existing docker image in the tests in `integration/test_create_dss_docker_image.py` simply add CLI option `--dss-docker-image` when calling `pytest`:
+
+```shell
+poetry run pytest --dss-docker-image exasol/data-science-sandbox:0.1.0
+```
```

Comment:
The section about tests I think is general and not specific to AWS. 
I propose to move back into general developer_guide or somewhere else.

Additionally 
* The directory `test/aws` has been moved to `integration/aws`.
* File `test/integration/test_ci.py` has been moved to `integration/codebuild`.

```suggestion
### Tests

XAL comes with a number of tests in directory `test`. Besides unit and integrations tests in the respective directories there are tests in directory `codebuild`, see [Executing AWS CodeBuild](#executing-aws-codebuild).  To run these tests manually please use
 ```shell export DSS_RUN_CI_TEST=true; poetry run test/codebuild/test_ci.py  ```

Creating a docker image is quite time-consuming, currently around 7 minutes. In order to use an existing docker image in the tests in `integration/test_create_dss_docker_image.py` simply add CLI option `--dss-docker-image` when calling `pytest`:

 ```shell  poetry run pytest --dss-docker-image exasol/data-science-sandbox:0.1.0  ```
```